packages/cid-verifier/src/bindings.d.ts

```diff
@@ -0,0 +1,58 @@
+import Toucan from "toucan-js";
+import { Logging } from "@web3-storage/worker-utils/loki";
+
+export {};
+
+// CF Analytics Engine types not available yet
+export interface AnalyticsEngine {
+  writeDataPoint(event: AnalyticsEngineEvent): void;
+}
+
+export interface AnalyticsEngineEvent {
+  readonly doubles?: number[];
+  readonly blobs?: (ArrayBuffer | string | null)[];
+}
+
+export interface EnvInput {
+  ENV: string;
+  DEBUG: string;
+  IPFS_GATEWAYS: string;
+  GATEWAY_HOSTNAME: string;
+  GOOGLE_CLOUD_API_URL: string;
+  GOOGLE_CLOUD_API_KEY: string;
+  EDGE_GATEWAY_API_URL: string;
+  REQUEST_TIMEOUT?: number;
+  SENTRY_DSN?: string;
+  LOKI_URL?: string;
+  LOKI_TOKEN?: string;
+  DENYLIST: KVNamespace;
+  GOOGLE_MALWARE_RESULTS: KVNamespace;
```

Comment:
Wondering if we should call this: `CID_VERIFIER`, so that we are better prepared if we had other tools. What do you think?

We did not discuss this yet, but looking at the code remembered we should consider this.

Let's consider that `sourceName` is the source where we get a result (in this case a google API code, the URL would be the nicest, but probably makes the KV key larger than the maximum allowed?). There are two options for how to setup `{key,value}`:
- key = `cid`, value = `[{ source: string, value: any }]`
- key = `cid/${sourceName}`, value  = sourceValue

The first option might be problematic for multiple atomic writes, given [KV is eventually consistent](https://developers.cloudflare.com/workers/learning/how-kv-works/).

The second option does not have write issues, but makes reads harder. We can [list by prefix](https://developers.cloudflare.com/workers/runtime-apis/kv/#listing-keys) (in this case CID) to get all the keys. There is an extra trick we could do, just have the response from the source being both KV value and metadata. If you see kv.list API, they return key + metadata, so we don't need list + n get operations to get what we care about. `metadata` can only have "The serialized JSON representation of the metadata object must be no more than 1024 bytes in length." This should be enough right? (we should document this)

Let me know what you think here @jsdevel  :)

Reply:
i kinda like a seperate KV named GOOGLE_MALWARE_RESULTS.  i went with cid for the key because the URL isn't really important once we know the cid is bad.  i think 1024 bytes will be enough in the value

Reply:
What would be your idea for when we add other services that are not google?

We still need to tackle the KV eventually consistent problem too. Any thoughts on that?